File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    sumset = set()  
    n = len(elements)
    elements = elements + elements  

    for i in range(n): 
        sum_value = 0
        for j in range(1, n + 1):  
            sum_value += elements[i + j - 1]
            sumset.add(sum_value)  

    return len(sumset)  

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const answer = new Set();
    const n = elements.length;
    
    for (let i = 0; i < n; i++) {
        let sum = 0;
        for (let j = 0; j < n; j++) {
            sum += elements[(i + j) % n];
            answer.add(sum);
        }
    }

    return answer.size; 
}
